RIOT-OS/RIOT
# net/sock/dtls: allow using multiple credentials

### Contribution description
Currently only one credential can be associated to each DTLS sock. This limits certain use cases that need to establish multiple connections on a single sock using potentially multiple credentials.

This PR proposes an extension of the sock DTLS API to allow registering multiple credentials to be used on a single sock. An application may register callbacks to specify a certain credential from the list based on the session information (and, when using PSK, based on the hint that may have been sent).

Additionally, now a server can send a PSK Identity hint to a client during the handshake, to help deciding on the PSK Identity that should be used.

Finally, the example application has been extended to show these functionalities.

### Testing procedure
- The example application should be working both when using PSK and ECC (for more information enable the debug messages in the sock_dtls.c file).
- Check that the correct hint is being sent in PSK
- Check that the correct RPK is being used in ECC
- Read the documentation
- A single credential use case should still work seamlessly

### Issues/PRs references
None so far
